src/ekam/CppActionFactory.cpp

```diff
@@ -209,6 +209,12 @@ Promise<void> LinkAction::startTarget(
 
   std::string compiler = cxx == NULL ? "c++" : cxx;
 
+  // Assume that any spaces indicate compiler wrappers (e.g. distcc, ccache, etc).
```

Comment:
I've thought about this some more & I don't really see the issue. If you did `CXX="g++ -m32"`, that wouldn't build with Ekam today anyway because "g++ -m32" isn't an executable. Can you help me understand how this worsens the current situation?

Reply:
Hmm, you might be right at present. I'm probably thinking of times when I used `CXX=g++ -m32` with autotools or cmake or something. But it does appear that for that to work here, there would have to be some argument-splitting code which is currently missing. But if we wanted to support that in the future, then it'll clash with the logic you're adding...

I think there's two clean options here:
* If the compiler wrapper can safely be used when linking, then we should use it. That is, parse `CXX` on whitespace, assume the first word is the executable and subsequent words are arguments that should be passed to it. Then this does exactly the right thing for both `g++ -m32` and when a compiler wrapper is in use.
* If the compiler wrapper really _shouldn't_ be used for linking, then I think it needs to be specified by a different flag, e.g. `CXX_WRAPPER`, which is prefixed to compiler steps in `compile.ekam-rule`. Or alternatively, maybe there should be a different flag to use to override the linker, like `CXX_LD`, which defaults to `$CXX` if not set explicitly.

Reply:
I tried option one but the challenge isn't the string splitting but that compiler name handling does special things to try to support cross compilation that are not very compatible with parsing white space. It would work in the case of args but not in the case of a wrapper since which substring is the compiler is ambiguous (could technically be any argument). In the totally abstract case, "clang" is also a valid string in a compiler arg  (e.g. `-Xclang`, `-I.../clang/...`, etc). So the general whitespace splitting approach is actually incompatible with supporting a wrapper correctly anyway. The current approach worked because it went the other way & just assumed the last space-delimited string is the compiler name which solves the "but an argument could contain the substring clang" case.

Given the available options, it seems like option 2 is really the only path forward. Would using the `LD` environment variable instead of `CXX` be more appropriate? That seems more "autotools"y & "pretty" but isn't back compatible. Also the code might break if `ld` is set to the real linker instead of the C++ frontend due to the linker name changing behavior so `LD` isn't even immediately clearly the right flag since the use-case is different enough from what autotools does.

Given these constraints, CXX_WRAPPER really feels like the path forward but let me know if you have any other thoughts.